### DISCUSSION TITLE
Need alittle help with this plugin :)

### DISCUSSION THREAD

**Discussion by jordangerada1** [dimonovdd/Xamarin.MediaGallery#30] [Q&A] [answered]
Hey guys I have recently tried to add this plugin to my project. All I want to do is be able to save a QR code thats on my xaml page to my gallery.  I believe I have done all the permissions correctly for both android and IOS aswell as the manifest files. My behind code is as follows: 
async void Save_Code(object sender, EventArgs e)
        {
            var screenshot = await Screenshot.CaptureAsync();
            await MediaGallery.SaveAsync(MediaFileType.Image, await screenshot.OpenReadAsync(), "myScreenshot.png");
        } 
 When I run my android Emulator and press my botton "Save_Code" I get this exception: 

System.IO.DirectoryNotFoundException: 'Could not find a part of the path "/storage/emulated/0/Pictures/MyAppBuild.Android/myScreenshot_20210517_185634.png".'

I honestly have no idea why its doing this but it has me stuck! Has anyone expereinced this problem before and knows how to fix it?? or could this be a potential bug with Android? Also I do appologise for my lack of knowledge, I just recented decided I wanted to learn Xamarin forms for my school project and my learning curve is very steep so any help would be greatly appreciated ! :)

**Comment by dimonovdd** [comment, marked as answer]
Hi. Have you requested permission from the user? You should get the `Granted` status.

```csharp
var status = await Permissions.RequestAsync<SaveMediaPermission>();

if (status != PermissionStatus.Granted)
    return;
 ```

**Reply by jordangerada1** [reply]
No I havent actually, I didnt think it was nessasary when saving. Let me run that code and I will get back to you! thank you for your speedy reply!

**Reply by jordangerada1** [reply]
Ok perfect its finally working and my screenshot was in the gallery. Thank you soo much for your help, i've learned something new today :)
